tools/builder/host/containers.py

```diff
@@ -0,0 +1,142 @@
+"""Code for managing the containers to build the repo."""
```

Comment:
Is there a reason this wasn't done with something like a Makefile? 
I have done this same exact thing where you use subprocess to run Docker. 

It's a freaking nightmare that is way easier to handle by just passing args to a Makefile command and running that directly. 

Otherwise, hitting the Unix Socket on the Docker REST API is way easier and more predictable than using subprocess.

Reply:
Mostly because I want to absolutely minimize the number of dependencies I need to do this. This way all you need is python. Interacting with the docker socket directly could be doable but again either i'd need to vendor it or handcode it, since I want `./build` to work without anything but python, and that doesn't really fit the timeline.

Plus, what we actually need out of `docker` here is really simple.

Inside the container it's going to be pretty gross too, but there I really don't want to deal with extensive makefile infrastructure.